Documentation/project-docs/project-nuget-dependencies.md

```diff
@@ -0,0 +1,52 @@
+
+As a historical note, `project.json` files in CoreFX previously had floating dependencies and `project.lock.json` files were checked in to lock the version for repeatable package restore. This was changed during pull request [#4894](https://github.com/dotnet/corefx/pull/4894).
+
+Dependency version validation
+-----------------------------
+
+The dependencies in each CoreFX project.json file are validated by a few rules to ensure package versions across the repository stay in sync. These rules are in `dir.props`, as `ValidationPattern` items. Each `ValidationPattern` item includes a regular expression that matches a set of package identities and metadata describing which version is expected for that regular expression. The metadata must be one of these:
+
+* `ExpectedPrerelease`: a prerelease version is expected, but any major, minor, and patch versions are fine. For example, an `ExpectedVersion` of `rc-12345` is valid for `1.0.0-rc-12345`, `5.4.3-rc-12345`, or any other `*-rc-12345`
+* `ExpectedVersion`: the full version string needs to match exactly. An example is `1.0.0-prerelease`. 
+
+Errors you can expect from failed dependency version validation are like the following:
+
+    error : Dependency validation error: for System.IO 4.0.10-rc2-10000 in src\System.Collections\tests\project.json package prerelease is 'rc2-10000', but expected 'rc2-23604' for packages matching '^((System\..*)|(Microsoft\.CSharp)|(Microsoft\.NETCore.*)|(Microsoft\.Win32\..*)|(Microsoft\.VisualBasic))(?<!TestData)$'
+
+    error : Dependency validation error: for System.Linq 4.0.0-beta-* in src\System.Collections\tests\project.json package prerelease is 'beta-', but expected 'rc2-23604' for packages matching '^((System\..*)|(Microsoft\.CSharp)|(Microsoft\.NETCore.*)|(Microsoft\.Win32\..*)|(Microsoft\.VisualBasic))(?<!TestData)$'
+    error : Floating dependency detected: System.Linq 4.0.0-beta-* in src\System.Collections\tests\project.json
+
+    error : Dependency validation error: for xunit 2.0.0 in src\System.Collections\tests\project.json package version is '2.0.0' but expected '2.1.0' for packages matching '^xunit$'
+
+To fix these, you can manually modify the `project.json` files mentioned or automatically fix them using the `UpdateInvalidPackageVersions` target described in the next section.
+
+Upgrading a package dependency
+------------------------------
+
+To update a package that isn't validated by a rule, simply change the project.json file.
+
+To update a package that is validated, follow these steps:
+
+1. Edit `ValidationPattern` item(s) in `dir.props` in the CoreFX root.
+2. Run the dependency update target in the repository root using this command:
+
+        build.cmd /t:UpdateInvalidPackageVersions
+
+The `UpdateInvalidPackageVersions` target looks through all dependencies, using the validation rules to update any invalid versions. The build prints out which dependencies are updated and which project.json files are written.
```

Comment:
Please add a note about doing this update as an independent commit so it is more easily reviewed in isolation from any other changes.